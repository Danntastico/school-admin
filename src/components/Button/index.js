import React from "react";
import { Btn } from "./styles";
import PropTypes from "prop-types";

export const Button = ({ onClick, label = "Button", isDisabled }) => {
  return (
    <Btn onClick={onClick} disabled={isDisabled}>
      {label}
    </Btn>
  );
};
