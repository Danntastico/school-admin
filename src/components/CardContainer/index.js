import React from 'react';
import { Container } from './styles';
export const CardContainer = ({ children }) => {
  return <Container>{children}</Container>;
};
