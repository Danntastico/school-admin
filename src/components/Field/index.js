import React from "react";
import { Input, Label, FieldContainer } from "./styles";
export const Field = ({
  disabled = false,
  fieldType,
  handleInputChange,
  inputType,
  label,
  name,
  placeholder,
  value,
  children,
}) => {
  const fieldRender = () => {
    switch (fieldType) {
      case "textarea":
        return (
          <textarea
            autoComplete="off"
            disabled={disabled}
            name={name}
            id={name}
            onChange={handleInputChange}
            value={value}
          />
        );
      case "select":
        return (
          <select name={name} value={value} onChange={handleInputChange}>
            {children}
          </select>
        );
      default:
        return (
          <Input
            autoComplete="off"
            disabled={disabled}
            id={name}
            name={name}
            onChange={handleInputChange}
            placeholder={placeholder}
            type={inputType}
            value={value}
            required={true}
          />
        );
    }
  };
  return (
    <FieldContainer>
      <Label htmlFor={name}> {label}</Label>
      {fieldRender()}
    </FieldContainer>
  );
};
