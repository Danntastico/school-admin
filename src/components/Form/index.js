import React from 'react';
import { FormHeader, FormTitle, FormBody, FormFooter } from './styles';
export const Form = ({ children, title, onSubmit }) => {
  return (
    <form onSubmit={onSubmit}>
      <FormHeader>
        <FormTitle> {title} </FormTitle>
      </FormHeader>
      <FormBody>{children[0]}</FormBody>
      <FormFooter>{children[1]}</FormFooter>
    </form>
  );
};
