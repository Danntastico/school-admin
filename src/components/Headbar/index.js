import React from 'react';

export const Headbar = ({ title }) => {
  return (
    <header>
      <h3>title</h3>
    </header>
  );
};
