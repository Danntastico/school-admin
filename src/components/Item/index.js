import React from "react";
import { ItemList } from "./styles";

export const Item = ({ children, bgColor, txtColor, to }) => (
  <ItemList to={to} bgcolor={bgColor} txtcolor={txtColor}>
    {children}
  </ItemList>
);
