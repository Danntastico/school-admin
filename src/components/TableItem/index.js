import React from 'react';
import { MdAccountCircle, MdLibraryBooks } from 'react-icons/md';

import { ItemContainer } from './styles';

const Icon = (type) => {
  switch (type) {
    case 'subject':
      return <MdLibraryBooks />;
    default:
      return <MdAccountCircle />;
  }
};

const testConfig = {
  id: '12345',
  lastName: 'lastName',
  name: 'name',
};

export const TableItem = ({ config = testConfig, type = 'teacher' }) => {
  const icon = Icon(type);
  const labels = Object.values(config);

  return (
    <ItemContainer>
      {icon}
      {labels.map((label) => (
        <p key={label}>{label}</p>
      ))}
      <button>Toggle Button</button>
    </ItemContainer>
  );
};
