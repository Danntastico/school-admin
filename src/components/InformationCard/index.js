import React from "react";
import { MdEdit } from "react-icons/md";
import { InfoCard, InfoHead, InfoBody, InfoTitle } from "./styles";
import { EditBtn } from "../Form/styles";
export const InformationCard = ({
  children,
  handleClickOnEdit,
  title,
  isEditable,
}) => {
  return (
    <InfoCard>
      <InfoHead>
        <InfoTitle>{title}</InfoTitle>
        {isEditable && (
          <EditBtn onClick={handleClickOnEdit}>
            <MdEdit />
          </EditBtn>
        )}
      </InfoHead>
      <InfoBody>{children}</InfoBody>
    </InfoCard>
  );
};
