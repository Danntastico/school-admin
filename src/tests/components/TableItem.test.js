import React from 'react';
import { shallow } from 'enzyme';
import { TableItem } from '../../components/TableItem';

describe('Unit tests for <TableItem />', () => {
  const testConfig = {
    id: '12345',
    lastName: 'lastName',
    name: 'name',
  };
  const wrapper = shallow(<TableItem config={testConfig} type='teacher' />);

  test('Should show the item correctly (shallow)', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
