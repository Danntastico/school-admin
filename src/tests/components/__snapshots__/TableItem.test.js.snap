// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for <TableItem /> Should show the item correctly (shallow) 1`] = `
<styled.div>
  <MdAccountCircle />
  <p
    key="12345"
  >
    12345
  </p>
  <p
    key="lastName"
  >
    lastName
  </p>
  <p
    key="name"
  >
    name
  </p>
  <button>
    Toggle Button
  </button>
</styled.div>
`;
